docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,197 @@
+= VirtualMachine Templates
+
+VirtualMachine templates are openshift templates (link:https://docs.openshift.org/latest/dev_guide/templates.html[Openshift Templates]) +
+that describe VirtualMachine specifications which produce the following template objects: +
+**One** OfflineVirtualMachine and **0-N** PersistentVolumeClaims. +
+The following document describes the process of creating a Virtual machine template which will be managed by ManageIQ.
+
+== Writing Templates
+You can define new VirtualMachine templates to make it easy to recreate all relevant objects (OfflineVirtualMachine and PersistentVolumeClaim). The template will define the objects it creates along with some metadata to guide the creation of those objects.
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha1
+kind: Template
+metadata:
+  name: RHEL-7 ❶
+  labels:
+    miq.github.io/kubvirt/is-vm-template: True
+objects:
+- apiVersion: kubevirt.io/v1alpha1
+  kind: OfflineVirtualMachine ❷
+  metadata:
+    name: rhel-7-${NAME} ❸
+  spec:
+    template:
+      metadata:
+        labels:
+          my: label
+      spec:
+        domain: 
+          resources: ❹
+            requests:
+              memory: ${MEMORY} 
+          cpu: ➎
+            cores: ${CPU}
+          devices:
+            disks: ➏
+              name: disk0
+              volumeName: mypvc 
+      volumes: ❼
+       - name: mypvc
+         presistentVolumeClaim:
+           claimName: rhel-7-pvc-${NAME}
+- apiVersion: kubevirt.io/v1alpha1
+  kind: PersistentVolumeClaim ❽
+  metadata:
+    name: rhel-7-pvc-${NAME} ➒
```

Comment:
here we need an annotation to say which volume this pvc should be based on

@aglitke do you have an example how a PVC should look to request a clone?

Reply:
The previous is a non blocking item for merging this - but we should then add a FIXME note

Reply:
@fabiand  added FIXME,  an issue can also be opened for that matter if needed

Reply:
Awesome, thank you!